dtfs-central-api/src/cron-scheduler-jobs/create-utilisation-reports/index.test.ts

```diff
@@ -28,29 +33,91 @@ describe('scheduler/jobs/create-utilisation-reports', () => {
     process.env = originalProcessEnv;
   });
 
-  describe('the task', () => {
-    const saveUtilisationReportSpy = jest.spyOn(UtilisationReportRepo, 'save');
-    const findOneByBankIdAndReportPeriodSpy = jest.spyOn(UtilisationReportRepo, 'findOneByBankIdAndReportPeriod');
-
-    const mockReportPeriod = {} as ReportPeriod;
-
-    const MONTHLY_REPORT_PERIOD_SCHEDULE = [
-      { startMonth: 1, endMonth: 1 },
-      { startMonth: 2, endMonth: 2 },
-      { startMonth: 3, endMonth: 3 },
-      { startMonth: 4, endMonth: 4 },
-      { startMonth: 5, endMonth: 5 },
-      { startMonth: 6, endMonth: 6 },
-      { startMonth: 7, endMonth: 7 },
-      { startMonth: 8, endMonth: 8 },
-      { startMonth: 9, endMonth: 9 },
-      { startMonth: 10, endMonth: 10 },
-      { startMonth: 11, endMonth: 11 },
-      { startMonth: 12, endMonth: 12 },
-    ];
+  let sendEmailSpy = jest.fn();
+
+  const saveUtilisationReportSpy = jest.spyOn(UtilisationReportRepo, 'save');
+  const findOneByBankIdAndReportPeriodSpy = jest.spyOn(UtilisationReportRepo, 'findOneByBankIdAndReportPeriod');
+
+  const MONTHLY_REPORT_PERIOD_SCHEDULE = [
+    { startMonth: 1, endMonth: 1 },
+    { startMonth: 2, endMonth: 2 },
+    { startMonth: 3, endMonth: 3 },
+    { startMonth: 4, endMonth: 4 },
+    { startMonth: 5, endMonth: 5 },
+    { startMonth: 6, endMonth: 6 },
+    { startMonth: 7, endMonth: 7 },
+    { startMonth: 8, endMonth: 8 },
+    { startMonth: 9, endMonth: 9 },
+    { startMonth: 10, endMonth: 10 },
+    { startMonth: 11, endMonth: 11 },
+    { startMonth: 12, endMonth: 12 },
+  ];
+
+  const mockBank = {
+    id: '123',
+    name: 'Test bank',
+    isVisibleInTfmUtilisationReports: true,
+    utilisationReportPeriodSchedule: MONTHLY_REPORT_PERIOD_SCHEDULE,
+  } as Bank;
+
+  const banks = [
+    {
+      ...mockBank,
+      id: '1',
+      name: 'Bank 1',
+    },
+    {
+      ...mockBank,
+      id: '2',
+      name: 'Bank 2',
+    },
+    {
+      ...mockBank,
+      id: '3',
+      name: 'Bank 3',
+    },
+  ] as Bank[];
+
+  const mockReportPeriod = {} as ReportPeriod;
+
+  describe('errors on save', () => {
+    beforeEach(() => {
+      jest.mocked(getAllBanks).mockResolvedValue(banks);
+      sendEmailSpy = jest.fn(() => Promise.resolve({}));
+      externalApi.sendEmail = sendEmailSpy;
+    });
+
+    it('should send an email when UtilisationReportRepo.save fails', async () => {
```

Comment:
Would it be worth simplifying this to just mock the implementation of UtilisationReportRepo.save to reject a promise rather than mocking the findOneByBankIdAndReportPeriodSpy to have custom behaviour to cause it to fail (which is then intertwined with the logic inside the .save function)?
line 95-102 could then just be a single mock on the .save function to make it reject

Reply:
changed and had to move the test to the bottom

Reply:
Thank you, looks cleaner! Re moving test to the bottom does that imply we aren't resetting our mocks properly and need to update them?